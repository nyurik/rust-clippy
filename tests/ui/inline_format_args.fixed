// run-rustfix

#![allow(clippy::eq_op)]
#![allow(clippy::format_in_format_args)]
#![allow(clippy::print_literal)]
#![allow(named_arguments_used_positionally)]
#![allow(unused_variables)]
#![warn(clippy::inline_format_args)]

fn tester(fn_arg: i32) {
    let local_i32 = 1;
    let local_f64 = 2.0;
    let local_opt: Option<i32> = Some(3);
    let width = 4;
    let prec = 5;
    let val = 6;

    // make sure this file hasn't been corrupted with tabs converted to spaces
    // TODO: ideally this should be a const compiler check, similar to
    // https://docs.rs/static_assertions/1.1.0/static_assertions/macro.const_assert_eq.html
    assert_eq!("	 	", "\t \t");

    println!("val='{local_i32}'");
    println!("val='{local_i32}'");
    println!("val='{local_i32}'");
    println!("val='{local_i32}'");
    println!("val='{local_i32}'");
    println!(
        "val='{local_i32}'"
    );
    println!("{local_i32}");
    println!("{fn_arg}");
    println!("{local_i32:?}");
    println!("{local_i32:#?}");
    println!("{local_i32:4}");
    println!("{local_i32:04}");
    println!("{local_i32:<3}");
    println!("{local_i32:#010x}");
    println!("{local_f64:.1}");
    println!("Hello {} is {local_f64:.*}", "x", local_i32);
    println!("{local_i32} {local_f64}");
    println!("{local_i32}, {}", local_opt.unwrap());
    println!("{val}");
    println!("{val}");

    // TODO: these cases should be handled
    println!("val='{\t }'", local_i32);
    println!("val='{\n }'", local_i32);
    println!("val='{local_i32}'", local_i32 = local_i32);
    println!("val='{local_i32}'", local_i32 = fn_arg);
    println!("{0}", local_i32);
    println!("{0:?}", local_i32);
    println!("{0:#?}", local_i32);
    println!("{0:04}", local_i32);
    println!("{0:<3}", local_i32);
    println!("{0:#010x}", local_i32);
    println!("{0:.1}", local_f64);
    println!("{0} {0}", local_i32);
    println!("{1} {} {0} {}", local_i32, local_f64);
    println!("{0} {1}", local_i32, local_f64);
    println!("{1} {0}", local_i32, local_f64);
    println!("{1} {0} {1} {0}", local_i32, local_f64);
    println!("{1} {0}", "str", local_i32);
    println!("{v}", v = local_i32);
    println!("{local_i32:0$}", width);
    println!("{local_i32:w$}", w = width);
    println!("{local_i32:.0$}", prec);
    println!("{local_i32:.p$}", p = prec);
    println!("{:0$}", width);
    println!("{:0$}", v = val);
    println!("{0:0$}", v = val);
    println!("{:0$.0$}", v = val);
    println!("{0:0$.0$}", v = val);
    println!("{0:0$.v$}", v = val);
    println!("{0:v$.0$}", v = val);
    println!("{v:0$.0$}", v = val);
    println!("{v:v$.0$}", v = val);
    println!("{v:0$.v$}", v = val);
    println!("{v:v$.v$}", v = val);
    println!("{:w$}", w = width);
    println!("{:.0$}", prec);
    println!("{:.p$}", p = prec);
    println!("{:0$.1$}", width, prec);
    println!("{:0$.w$}", width, w = prec);
    println!("{:1$.2$}", local_f64, width, prec);
    println!("Width = {}, value with width = {:0$}", local_i32, local_f64);
    println!("{:p$.w$}", local_i32, w = width, p = prec);
    println!("{:p$.w$}", p = width, w = prec);
    println!("{}", format!("{}", local_i32));

    // wouldn't be modified by the lint
    println!(concat!("nope ", "{}"), local_i32);
    println!("val='{local_i32}'");
    println!("val='{local_i32 }'");
    println!("val='{local_i32	}'"); // with tab
    println!("val='{local_i32\n}'");
    println!("{}", usize::MAX);
    println!("{}", local_opt.unwrap());
    println!(
        "val='{local_i32
    }'"
    );
}

fn main() {
    tester(42);
}
