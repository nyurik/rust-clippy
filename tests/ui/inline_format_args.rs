// run-rustfix

#![allow(clippy::eq_op)]
#![allow(clippy::format_in_format_args)]
#![allow(clippy::print_literal)]
#![allow(named_arguments_used_positionally)]
#![allow(unused_variables)]
#![warn(clippy::inline_format_args)]

fn tester(fn_arg: i32) {
    let local_value = 1;
    let local_value2 = 2.0;
    let local_opt: Option<i32> = Some(3);
    let width = 4;
    let prec = 5;
    let val = 6;

    // make sure this file hasn't been corrupted with tabs converted to spaces
    // TODO: ideally this should be a const compiler check, similar to
    // https://docs.rs/static_assertions/1.1.0/static_assertions/macro.const_assert_eq.html
    assert_eq!("	 	", "\t \t");

    println!("val='{}'", local_value);
    println!("val='{   }'", local_value);
    println!("val='{	}'", local_value);
    println!("val='{ 	}'", local_value);
    println!("val='{	 }'", local_value);
    println!(
        "val='{
    }'",
        local_value
    );
    println!("{}", local_value);
    println!("{}", fn_arg);
    println!("{:?}", local_value);
    println!("{:#?}", local_value);
    println!("{:4}", local_value);
    println!("{:04}", local_value);
    println!("{:<3}", local_value);
    println!("{:#010x}", local_value);
    println!("{:.1}", local_value2);
    println!("Hello {} is {:.*}", "x", local_value, local_value2);
    println!("{} {}", local_value, local_value2);
    println!("{}, {}", local_value, local_opt.unwrap());
    println!("{}", val);
    println!("{}", v = val);

    // TODO: these cases should be handled
    println!("val='{\t }'", local_value);
    println!("val='{\n }'", local_value);
    println!("val='{local_value}'", local_value = local_value);
    println!("val='{local_value}'", local_value = fn_arg);
    println!("{0}", local_value);
    println!("{0:?}", local_value);
    println!("{0:#?}", local_value);
    println!("{0:04}", local_value);
    println!("{0:<3}", local_value);
    println!("{0:#010x}", local_value);
    println!("{0:.1}", local_value2);
    println!("{0} {0}", local_value);
    println!("{1} {} {0} {}", local_value, local_value2);
    println!("{0} {1}", local_value, local_value2);
    println!("{1} {0}", local_value, local_value2);
    println!("{1} {0} {1} {0}", local_value, local_value2);
    println!("{1} {0}", "str", local_value);
    println!("{v}", v = local_value);
    println!("{local_value:0$}", width);
    println!("{local_value:w$}", w = width);
    println!("{local_value:.0$}", prec);
    println!("{local_value:.p$}", p = prec);
    println!("{:0$}", width);
    println!("{:0$}", v = val);
    println!("{0:0$}", v = val);
    println!("{:0$.0$}", v = val);
    println!("{0:0$.0$}", v = val);
    println!("{0:0$.v$}", v = val);
    println!("{0:v$.0$}", v = val);
    println!("{v:0$.0$}", v = val);
    println!("{v:v$.0$}", v = val);
    println!("{v:0$.v$}", v = val);
    println!("{v:v$.v$}", v = val);
    println!("{:w$}", w = width);
    println!("{:.0$}", prec);
    println!("{:.p$}", p = prec);
    println!("{:0$.1$}", width, prec);
    println!("{:0$.w$}", width, w = prec);
    println!("{:1$.2$}", local_value2, width, prec);
    println!("Width = {}, value with width = {:0$}", local_value, local_value2);
    println!("{:p$.w$}", local_value, w = width, p = prec);
    println!("{:p$.w$}", p = width, w = prec);
    println!("{}", format!("{}", local_value));

    // wouldn't be modified by the lint
    println!(concat!("nope ", "{}"), local_value);
    println!("val='{local_value}'");
    println!("val='{local_value }'");
    println!("val='{local_value	}'"); // with tab
    println!("val='{local_value\n}'");
    println!("{}", usize::MAX);
    println!("{}", local_opt.unwrap());
    println!(
        "val='{local_value
    }'"
    );
}

fn main() {
    tester(42);
}
