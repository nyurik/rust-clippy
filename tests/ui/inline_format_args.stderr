error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:23:5
   |
LL |     println!("val='{}'", local_i32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::inline-format-args` implied by `-D warnings`
help: change this to
   |
LL -     println!("val='{}'", local_i32);
LL +     println!("val='{local_i32}'");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:24:5
   |
LL |     println!("val='{   }'", local_i32); // 3 spaces
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("val='{   }'", local_i32); // 3 spaces
LL +     println!("val='{local_i32}'"); // 3 spaces
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:25:5
   |
LL |     println!("val='{    }'", local_i32); // tab
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("val='{    }'", local_i32); // tab
LL +     println!("val='{local_i32}'"); // tab
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:26:5
   |
LL |     println!("val='{     }'", local_i32); // space+tab
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("val='{     }'", local_i32); // space+tab
LL +     println!("val='{local_i32}'"); // space+tab
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:27:5
   |
LL |     println!("val='{     }'", local_i32); // tab+space
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("val='{     }'", local_i32); // tab+space
LL +     println!("val='{local_i32}'"); // tab+space
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:28:5
   |
LL | /     println!(
LL | |         "val='{
LL | |     }'",
LL | |         local_i32
LL | |     );
   | |_____^
   |
help: change this to
   |
LL -         "val='{
LL +         "val='{local_i32}'"
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:33:5
   |
LL |     println!("{}", local_i32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{}", local_i32);
LL +     println!("{local_i32}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:34:5
   |
LL |     println!("{}", fn_arg);
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{}", fn_arg);
LL +     println!("{fn_arg}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:35:5
   |
LL |     println!("{:?}", local_i32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:?}", local_i32);
LL +     println!("{local_i32:?}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:36:5
   |
LL |     println!("{:#?}", local_i32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:#?}", local_i32);
LL +     println!("{local_i32:#?}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:37:5
   |
LL |     println!("{:4}", local_i32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:4}", local_i32);
LL +     println!("{local_i32:4}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:38:5
   |
LL |     println!("{:04}", local_i32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:04}", local_i32);
LL +     println!("{local_i32:04}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:39:5
   |
LL |     println!("{:<3}", local_i32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:<3}", local_i32);
LL +     println!("{local_i32:<3}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:40:5
   |
LL |     println!("{:#010x}", local_i32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:#010x}", local_i32);
LL +     println!("{local_i32:#010x}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:41:5
   |
LL |     println!("{:.1}", local_f64);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:.1}", local_f64);
LL +     println!("{local_f64:.1}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:45:5
   |
LL |     println!("{} {}", local_i32, local_f64);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{} {}", local_i32, local_f64);
LL +     println!("{local_i32} {local_f64}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:46:5
   |
LL |     println!("{}, {}", local_i32, local_opt.unwrap());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{}, {}", local_i32, local_opt.unwrap());
LL +     println!("{local_i32}, {}", local_opt.unwrap());
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:47:5
   |
LL |     println!("{}", val);
   |     ^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{}", val);
LL +     println!("{val}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:48:5
   |
LL |     println!("{}", v = val);
   |     ^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{}", v = val);
LL +     println!("{val}");
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:85:5
   |
LL |     println!("{:w$}", local_i32, w = width);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:w$}", local_i32, w = width);
LL +     println!("{local_i32:w$}", w = width);
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:89:5
   |
LL |     println!("{:.p$}", local_i32, p = prec);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:.p$}", local_i32, p = prec);
LL +     println!("{local_i32:.p$}", p = prec);
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:102:5
   |
LL |     println!("{:p$.w$}", local_i32, w = width, p = prec);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{:p$.w$}", local_i32, w = width, p = prec);
LL +     println!("{local_i32:p$.w$}", w = width, p = prec);
   |

error: variables can be used directly in the `format!` string
  --> $DIR/inline_format_args.rs:104:20
   |
LL |     println!("{}", format!("{}", local_i32));
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change this to
   |
LL -     println!("{}", format!("{}", local_i32));
LL +     println!("{}", format!("{local_i32}"));
   |

error: aborting due to 23 previous errors

